api/v1beta1/watcherapplier_types.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// WatcherApplierSpec defines the desired state of WatcherApplier
+type WatcherApplierSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of WatcherApplier. Edit watcherapplier_types.go to remove/update
+	Foo string `json:"foo,omitempty"`
```

Comment:
so is the plan to merge this skeleton and then work on filling out each service?

or will you start iterating here

Reply:
That's my proposal, yes. Merge the empty skeleton and then filling out each service.